### PR TITLE
feat(Popover): new prop unstable_disableAutoFocus to prevent auto focus on Popover open

### PR DESCRIPTION
**Pull request by YuanboXue-Amber**
When Popover opens, it automatically focuses on first focusable item in PopoverSurface.
There's no way to disable this from user side. After discussion we decide to introduce unstable prop `unstable_disableAutoFocus`

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/290fb885246f86efa306c9bb65ff17107b390b8b'>290fb885246f86efa306c9bb65ff17107b390b8b</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26141/perf-test-react-components/Avatar-mount_base.html">1527</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26141/perf-test-react-components/Avatar-mount.html">1554</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26141/perf-test-react-components/Button-mount_base.html">1131</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26141/perf-test-react-components/Button-mount.html">1108</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26141/perf-test-react-components/FluentProvider-mount_base.html">1838</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26141/perf-test-react-components/FluentProvider-mount.html">1891</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26141/perf-test-react-components/FluentProviderWithTheme-mount_base.html">750</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26141/perf-test-react-components/FluentProviderWithTheme-mount.html">740</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26141/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">682</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26141/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">716</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26141/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">735</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26141/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">746</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26141/perf-test-react-components/MakeStyles-mount_base.html">2260</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26141/perf-test-react-components/MakeStyles-mount.html">2270</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26141/perf-test-react-components/Persona-mount_base.html">3562</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26141/perf-test-react-components/Persona-mount.html">3492</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26141/perf-test-react-components/SpinButton-mount_base.html">2954</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26141/perf-test-react-components/SpinButton-mount.html">2996</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-popover/src/components/Popover/usePopover.ts [hunk 1]
```diff
@@ -120,14 +120,14 @@ export const usePopover_unstable = (props: PopoverProps): PopoverState => {
   const { findFirstFocusable } = useFocusFinders();
 
   React.useEffect(() => {
-    if (open && positioningRefs.contentRef.current) {
+    if (open && positioningRefs.contentRef.current && !props.unstable_disableAutoFocus) {
```
Comment (layershifter) [inline packages/react-components/react-popover/src/components/Popover/usePopover.ts:123]:
```suggestion
    if (props.unstable_disableAutoFocus) {
      return;
    }

    if (open && positioningRefs.contentRef.current) {
```

nit: can you please do an early return instead?

**Review by layershifter — APPROVED** [review]

**Comment by ling1726** [comment]
would also be nice to add a test in cypress

**Pull request merged**